src/test/java/pl/ostek/internet_chat/controller/MessageControllerIntegrationTest.java

```diff
@@ -0,0 +1,79 @@
+package pl.ostek.internet_chat.controller;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.ResultActions;
+import pl.ostek.internet_chat.InternetChatApplication;
+import pl.ostek.internet_chat.model.Message;
+import pl.ostek.internet_chat.repository.MessageRepository;
+
+import java.util.Arrays;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.hamcrest.Matchers.hasSize;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.*;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.MOCK,
+        classes = InternetChatApplication.class)
+@AutoConfigureMockMvc
+public class MessageControllerIntegrationTest {
```

Comment:
Przydaje się mieć przynajmniej po jednym teście pełnego requesta w oparciu o json zapisany tekstem (albo wprost w teście, albo jako plik z resource-ów np. getResource/getResourceAsStream) i podobnie z weryfikacją response - zamiast jsonPath, albo porównania z jsonem generowanym objectMapperem, porównać z tekstem json.
To trochę droga na "skróty" - można testować każdy aspekt danych osobnym testem i assertami andExpect(jsonPath... ale przy dużych komunikatach request/response - to dość sporo testów :)
Natomiast ObjectMapper może być w różny sposób przekonfigurowywany i użycie tekstu json zapewnia w większym stopniu zgodność z kontraktem dla danego endpointa.

Testy jak napisałeś - jak najbardziej ok :) Ale warto "przećwiczyć" w testach integracyjnych też podejście z pełnym requestem/responsem z pliku z test-resource-ów lub (dla małych komunikatów) osadzonych w teście jako tekst :)